sdk/core/azure-core/azure/core/_pipeline_client_async.py

```diff
@@ -189,6 +209,7 @@ async def _make_pipeline_call(self, request, **kwargs):
                     # the body is loaded. instead of doing response.read(), going to set the body
                     # to the internal content
                     rest_response._content = response.body()  # pylint: disable=protected-access
+                    await rest_response.read()
```

Comment:
I'm not sure I quite follow the comment above... If the content is already read, why do we need to call `rest_response.read()`? I know there be dragons here, but this feels a bit convoluted and error prone....

Reply:
yeah should've called it out: this is to make sure that all of the attributes we set in `def read()` are correctly set. This code is going to be outdated in the "move pipelines to rest PR", but needed to fix it in this PR to pass tests

Reply:
It would be better if there was a temporary setter (or helper method) on `rest_response` that would handle all of its internal state. I understand that the intent for this code is to be temporary, but I'm old and cynical and I am pretty sure that there is temporary code that I wrote back in '03 that still ships in a product.

Reply:
no problem, will add that!